chia/wallet/wallet_node.py

```diff
@@ -225,6 +232,65 @@ async def get_private_key(self, fingerprint: Optional[int]) -> Optional[PrivateK
 
         return key
 
+    def set_resync_flag(self, fingerprint, flag: bool = True):
+        with lock_and_load_config(self.root_path, "config.yaml") as config:
+            if flag is True:
+                config["wallet"]["reset_sync_for_fingerprint"] = fingerprint
+                self.log.info("Enabled resync for wallet fingerprint: %s", fingerprint)
+            else:
+                self.log.debug(
+                    "Trying to disable resync: %s [%s]", fingerprint, config["wallet"].get("reset_sync_for_fingerprint")
+                )
+                if config["wallet"].get("reset_sync_for_fingerprint") == fingerprint:
+                    del config["wallet"]["reset_sync_for_fingerprint"]
+                    self.log.info("Disabled resync for wallet fingerprint: %s", fingerprint)
+            save_config(self.root_path, "config.yaml", config)
+
+    async def reset_sync_db(self, db_path, fingerprint):
+        conn: aiosqlite.Connection
+        async with manage_connection(db_path) as conn:
+            self.log.info("Resetting wallet sync data...")
+            rows = list(await conn.execute_fetchall("SELECT name FROM sqlite_master WHERE type='table'"))
+            try:
+                # if less than 20 it means we're running on old database and that's
+                # handled, but not if there's more
+                assert len(rows) < 20, (
```

Comment:
I don't think this is an appropriate use of `assert`. At least in my book, `assert` is meant to check invariants that are supposed to be maintained by the program, and them being violated is a bug. This is not that.

It would seem more appropriate to just *check* the number of tables and if they're fewer than 20, print the error log below and return.

I also think it would make sense to add some details to this check, it doesn't seem super reliable. What if we optimize away a few tables in the future? Will we remember to change this number?